src/rpc/mining.cpp

```diff
@@ -1064,7 +1064,13 @@ static RPCHelpMan estimatesmartfee()
 {
     RPCTypeCheck(request.params, {UniValue::VNUM, UniValue::VSTR});
     RPCTypeCheckArgument(request.params[0], UniValue::VNUM);
-    unsigned int max_target = ::feeEstimator.HighestTargetTracked(FeeEstimateHorizon::LONG_HALFLIFE);
+
+    NodeContext& node = EnsureNodeContext(request.context);
+    if (!node.feeEstimator) {
+        throw JSONRPCError(RPC_INTERNAL_ERROR, "Fee estimation disabled");
+    }
```

Comment:
How about extracting this to an `EnsureFeeEstimator()` function? (see `EnsureMemPool` and `EnsureChainman`)

Reply:
I thought `EnsureXX()` were used for components that should not be absent, while the fee estimator may be ?

Reply:
I think it's fine to add `EnsureXX()` for components that may not be enabled. Does that sound right to you @MarcoFalke ?

Reply:
It doesn't make sense to call `estimatesmartfee` without the fee estimator. So yes, `EnsureFeeEstimator` is the way to go right now. (In the future the fee estimator can register the RPC itself, so when it is missing, the RPC won't even be there)

Reply:
> In the future the fee estimator can register the RPC itself, so when it is missing, the RPC won't even be there

Hmm that would be particularly nice, as for example applications up the stack relying on fee estimation would be able to just error at startup rather than "check and see if it errors". Might propose a follow-up implementing this.